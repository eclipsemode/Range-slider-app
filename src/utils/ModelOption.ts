declare type ModelOption = {
    from: number,
    to: number,
    gap: number,
    step: number,
    min: number,
    max: number,
    vertical: boolean;
    rulers: boolean,
    range: boolean,
    tooltip: boolean,
    percent: boolean,
    color: {
        firstColor: string,
        secondColor: string,
        textColor: string,
        thumbColor: string
    },
    configPanel: boolean,
    controlConfig: string[],
    toggleConfig: string[],
    progress: boolean
}

export default ModelOption;