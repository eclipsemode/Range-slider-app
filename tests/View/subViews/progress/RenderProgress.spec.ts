import $ from "jquery";
import {
  CreateBar,
  CreateProgress,
  RenderProgress,
} from "../../../../src/View/subViews";
import ModelOption from "../../../../src/utils/ModelOption";

describe("Render Progress test", () => {
  const body: JQuery = $("body");
  const options: ModelOption = {
    max: 104,
    tooltip: true,
    range: true,
    configPanel: true,
    step: 33,
    gap: 1,
    min: 0,
    from: 0,
    to: 99,
    vertical: false,
    rulers: true,
    percent: false,
    color: {
      firstColor: "#ffe53b",
      secondColor: "#ff2525",
      textColor: "#ffe53b",
      thumbColor: "#ffe53b",
    },
    progress: true,
    controlConfig: ["min", "max", "step", "from", "to"],
    toggleConfig: ["vertical", "range", "rulers", "progress", "tooltip"],
  };
  const progress: CreateProgress = new CreateProgress(body);
  const bar: CreateBar = new CreateBar(body);

  test("Should be defined", () => {
    const renderProgress: RenderProgress = new RenderProgress(
      options,
      progress,
      bar
    );
    expect(renderProgress.progress).toBeDefined();
  });
  test("Should match snapshot", () => {
    const renderProgress: RenderProgress = new RenderProgress(
      options,
      progress,
      bar
    );
    expect(renderProgress.progress.progressElement).toMatchSnapshot(
      "slider-app__progress"
    );
  });
  test("Should have length 1", () => {
    const renderProgress: RenderProgress = new RenderProgress(
      options,
      progress,
      bar
    );
    expect(renderProgress.progress.progressElement).toHaveLength(1);
  });
  test("Should match vertical", () => {
    const renderProgress: RenderProgress = new RenderProgress(
      { ...options, vertical: true },
      progress,
      bar
    );
    expect(renderProgress.progress.progressElement[0]).toMatchSnapshot(
      '<div class="slider-app__progress slider-app__progress--vertical" />'
    );
  });
  test("Should not to match element", () => {
    const renderProgress: RenderProgress = new RenderProgress(
      { ...options, progress: false },
      progress,
      bar
    );
    expect(renderProgress.progress).toBe(null);
  });
});
