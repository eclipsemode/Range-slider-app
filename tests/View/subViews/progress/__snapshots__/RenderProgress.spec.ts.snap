// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Progress test Should match snapshot: <div class="slider-app__progress"> 1`] = `
jQuery {
  "0": <body>
    <div
      class="slider-app__progress"
    />
    <div
      class="slider-app__bar"
    />
  </body>,
  "length": 1,
  "prevObject": jQuery {
    "0": Document {
      "location": Location {
        "assign": [Function],
        "hash": "",
        "host": "localhost",
        "hostname": "localhost",
        "href": "http://localhost/",
        "origin": "http://localhost",
        "pathname": "/",
        "port": "",
        "protocol": "http:",
        "reload": [Function],
        "replace": [Function],
        "search": "",
        "toString": [Function],
      },
      Symbol(SameObject caches): Object {
        "childNodes": NodeList [
          DocumentType {},
          <html>
            <head />
            <body>
              <div
                class="slider-app__progress"
              />
              <div
                class="slider-app__bar"
              />
            </body>
          </html>,
        ],
        "implementation": DOMImplementation {},
      },
    },
    "length": 1,
  },
}
`;

exports[`Render Progress test Should match snapshot: slider-app__progress 1`] = `
jQuery {
  "0": <div
    class="slider-app__progress"
  />,
  "length": 1,
}
`;

exports[`Render Progress test Should match vertical: <div class="slider-app__progress slider-app__progress--vertical" /> 1`] = `
<div
  class="slider-app__progress slider-app__progress--vertical"
/>
`;
